services/compute/virtualmachinescaleset/virtualmachinescaleset.go

```diff
@@ -134,6 +139,18 @@ func (c *client) getVirtualMachineScaleSetSecurityProfile(vm *wssdcloudcompute.V
 	return securityProfile, nil
 }
 
+func (c *client) getVirtualMachineScaleSetGuestAgentProfile(vm *wssdcloudcompute.VirtualMachineProfile) (*compute.GuestAgentProfile, error) {
```

Comment:
same here, always a nil

Reply:
The other functions also have the error as a returned value, so we added it too for consistency.